### PR TITLE
languages/astro: init

### PR DESCRIPTION
**Pull request by DamitusThyYeetus123**
Adds Astro to the list of languages supported. Also drops the shipped mnw version to fix an issue with node-based language servers.

## Sanity Checking

[editorconfig]: https://editorconfig.org
[changelog]: https://github.com/NotAShelf/nvf/tree/main/docs/release-notes

- [ ] I have updated the [changelog] as per my changes.
- [x] I have tested, and self-reviewed my code.
- Style and consistency
  - [ ] I ran **Alejandra** to format my code (`nix fmt`).
  - [x] My code conforms to the [editorconfig] configuration of the project.
  - [x] My changes are consistent with the rest of the codebase.
- If new changes are particularly complex:
  - [ ] My code includes comments in particularly complex areas
  - [ ] I have added a section in the manual.
  - [ ] _(For breaking changes)_ I have included a migration guide.
- Package(s) built:
  - [ ] `.#nix` (default package)
  - [x] `.#maximal`
  - [ ] `.#docs-html`
- Tested on platform(s)
  - [x] `x86_64-linux`
  - [ ] `aarch64-linux`
  - [ ] `x86_64-darwin`
  - [ ] `aarch64-darwin`

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: configuration.nix [hunk 1]
```diff
@@ -48,6 +48,7 @@ isMaximal: {
 
       # Assembly is not common, and the asm LSP is a major hit-or-miss
       assembly.enable = false;
+      astro.enable = isMaximal;
```
Comment (NotAShelf) [inline configuration.nix:51]:
```suggestion
      astro.enable = false;
```

It's not needed in maximal.

File: flake/develop.nix [hunk 2]
```diff
@@ -18,7 +18,8 @@
     # testing, but make sure to discard the changes before creating a pull
     # request.
     packages.dev = let
-      configuration = {};
+      configuration = {
+      };
```
Comment (NotAShelf) [inline flake/develop.nix:22]:
```suggestion
      configuration = {};
```

?

File: modules/plugins/languages/astro.nix [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+      server = mkOption {
+        description = "Astro LSP server to use";
+        type = enum (attrNames servers);
+        default = defaultServer;
+      };
+
+      package = mkOption {
+        description = "Astro LSP server package, or the command to run as a list of strings";
+        example = ''[lib.getExe pkgs.jdt-language-server "-data" "~/.cache/jdtls/workspace"]'';
```
Comment (NotAShelf) [inline modules/plugins/languages/astro.nix:101]: Use an example that mentions astro LSP, not jdt.

**Review by NotAShelf — CHANGES_REQUESTED** [review]
LGTM. Few changes, and requires a changelog entry.

File: flake/develop.nix (hunk 2 above)
Comment (DamitusThyYeetus123) [reply flake/develop.nix:22]: Formatter auto formatted, my bad

File: modules/plugins/languages/astro.nix (hunk 3 above)
Comment (DamitusThyYeetus123) [reply modules/plugins/languages/astro.nix:101]: Will do, tbh just copied the file from svelte.nix

**Pull request merged**